ftd/src/js/element.rs

```diff
@@ -699,6 +704,165 @@ impl Image {
+    pub fit: Option<ftd::js::Value>,
+    pub controls: Option<ftd::js::Value>,
+    pub loop_video: Option<ftd::js::Value>,
+    pub muted: Option<ftd::js::Value>,
+    pub autoplay: Option<ftd::js::Value>,
+    pub poster: Option<ftd::js::Value>,
+    pub common: Common,
+}
+
+impl Video {
+    pub fn from(component: &ftd::interpreter::Component) -> Video {
+        let component_definition = ftd::interpreter::default::default_bag()
+            .get("ftd#video")
+            .unwrap()
+            .clone()
+            .component()
+            .unwrap();
+        Video {
+            src: ftd::js::value::get_optional_js_value(
+                "src",
+                component.properties.as_slice(),
+                component_definition.arguments.as_slice(),
+            )
+            .unwrap(),
+            fit: ftd::js::value::get_optional_js_value(
+                "fit",
+                component.properties.as_slice(),
+                component_definition.arguments.as_slice(),
+            ),
+            autoplay: ftd::js::value::get_optional_js_value(
+                "autoplay",
+                component.properties.as_slice(),
+                component_definition.arguments.as_slice(),
+            ),
+            controls: ftd::js::value::get_optional_js_value(
+                "controls",
+                component.properties.as_slice(),
+                component_definition.arguments.as_slice(),
+            ),
+            loop_video: ftd::js::value::get_optional_js_value(
+                "loop",
+                component.properties.as_slice(),
+                component_definition.arguments.as_slice(),
+            ),
+            muted: ftd::js::value::get_optional_js_value(
+                "muted",
+                component.properties.as_slice(),
+                component_definition.arguments.as_slice(),
+            ),
+            poster: ftd::js::value::get_optional_js_value(
+                "poster",
+                component.properties.as_slice(),
+                component_definition.arguments.as_slice(),
+            ),
+            common: Common::from(
+                component.properties.as_slice(),
+                component_definition.arguments.as_slice(),
+                component.events.as_slice(),
+            ),
+        }
+    }
+
+    pub fn to_component_statements(
+        &self,
+        parent: &str,
+        index: usize,
+        doc: &ftd::interpreter::TDoc,
+        rdata: &mut ftd::js::ResolverData,
+        should_return: bool,
+    ) -> Vec<fastn_js::ComponentStatement> {
+        let mut component_statements = vec![];
+        let kernel = create_element(fastn_js::ElementKind::Video, parent, index, rdata);
+        component_statements.push(fastn_js::ComponentStatement::CreateKernel(kernel.clone()));
+        component_statements.push(fastn_js::ComponentStatement::SetProperty(
+            fastn_js::SetProperty {
+                kind: fastn_js::PropertyKind::ImageSrc,
```

Comment:
Change this to VideoSrc 
```suggestion
                kind: fastn_js::PropertyKind::VideoSrc,
```